dark/views/tournament/info.py

```diff
@@ -12,9 +15,24 @@ class TournamentInfoView(DetailView):
     slug_url_kwarg = 'tournament'
     slug_field = 'id'
 
-    def get_context_data(self, **kwargs):
-        context = super().get_context_data(**kwargs)
-        context['rounds'] = TournamentRound.objects.filter(tournament=self.object)
-        context['contestants'] = Team.objects.filter(tournament=self.object)
-        context['is_creator_viewing'] = self.object.creator == self.request.user
-        return context
+    def get(self, request, *args, **kwargs):
+        if not request.user.is_authenticated:
+            return redirect(reverse('tournament:all'))
+
+        self.object = get_object_or_404(Tournament, id=self.kwargs['tournament'])
+
+        tournament = self.object
+        user = request.user
+
+        if tournament.is_private and user not in tournament.participants.all():
```

Comment:
Jest to nieprzyjemne więć nie nalegam ale na to możnaby dopisać własnego Mixina do common/views

Reply:
Nie do końca wiem jak to robić i raczej zajmie to umnie sporo czasu, więc jabym to opuscił